shared/bridges/cctp/cosmos_bridge_client.go

```diff
@@ -171,11 +171,37 @@ func (c *CosmosBridgeClient) GetTxResult(ctx context.Context, txHash string) (*b
 			return nil, nil, ErrTxResultNotFound{TxHash: txHash}
 		}
 		return nil, nil, err
-	} else if result.TxResult.Code != 0 {
-		return big.NewInt(result.TxResult.GasUsed), &TxFailure{fmt.Sprintf("tx failed with code: %d and log: %s", result.TxResult.Code, result.TxResult.Log)}, nil
 	}
 
-	return big.NewInt(result.TxResult.GasUsed), nil, nil
+	// parse tx fee event and use as the gas cost. we are using the fee event
+	// as the gas cost, technically this is not always true for all cosmos
+	// txns, however for all of the transactions that the solver will submit
+	// and get results of via this function, this should be true
```

Comment:
do we want to throw an error if a fee event is not found?

Reply:
ye, added